tools/toolchains/__init__.py

```diff
@@ -433,7 +433,7 @@ def __init__(self, target, notify=None, macros=None, build_profile=None,
         self.build_all = False
 
         # Build output dir
-        self.build_dir = build_dir
+        self.build_dir = abspath(build_dir) if PRINT_COMPILER_OUTPUT_AS_LINK else build_dir
```

Comment:
Why is this change needed?

Reply:
Because the GCC gives out what he gets, i.e. when you are passing relative paths the gcc prints out relative paths and vice versa.